src/library/scala/collection/Map.scala

```diff
@@ -112,21 +100,18 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
     s.asInstanceOf[S]
   }
 
-  /**
-    * Type alias to `CC`. It is used to provide a default implementation of the `fromSpecific`
-    * and `newSpecificBuilder` operations.
-    *
-    * Due to the `@uncheckedVariance` annotation, usage of this type member can be unsound and is
-    * therefore not recommended.
-    */
-  protected type MapCC[KCC, VCC] = CC[KCC, VCC] @uncheckedVariance
-
   /** Similar to `fromIterable`, but returns a Map collection type.
-    * Note that the return type is now `CC[K2, V2]` aka `MapCC[K2, V2]` rather than `IterableCC[(K2, V2)]`.
+    * Note that the return type is now `CC[K2, V2]`.
     */
   @`inline` protected final def mapFromIterable[K2, V2](it: Iterable[(K2, V2)]): CC[K2, V2] = mapFactory.from(it)
 
-  def mapFactory: MapFactory[MapCC]
+  /** The companion object of this map, providing various factory methods.
+    *
+    * @note When implementing a custom collection type and refining `CC` to the new type, this
+    *       method needs to be overridden to return a factory for the new type (the compiler will
+    *       issue an error otherwise).
+    */
+  def mapFactory: MapFactory[CC]
```

Comment:
`Iterable.iterableFactory` has a default implementation so you can just `extends Iterable` in simple cases. It should probably be the same for `mapFactory`.

Reply:
This abstract method is in MapOps, the default implementation is in Map (https://github.com/scala/scala/pull/7929/files#diff-7e4a5aff40ea5ded3e138d24f2e3cacaR29). Or did I misunderstand?

Reply:
Oh, right. I blame it on the low contrast of the screen in the sun when working outdoors :-)